lib/spack/spack/database.py

```diff
@@ -393,7 +393,7 @@ def _add(self, spec, path, directory_layout=None):
 
         else:
             self._data[key] = InstallRecord(spec, path, True)
-            for dep in spec.dependencies.values():
+            for dep in spec.dependencies(spack.DEP_INSTALL):
```

Comment:
One reason to drop pure-build deps is that there's no reason to ship autoconf, CMake, py-setuptools, etc. with your final application's RPM.